src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,154 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+
+    // The relative path to the directory
+    private final String directory;
+
+    // The file
+    private final String file;
+
+    // The contents of the file as a List of Strings
+    private ArrayList<String> fileTasks;
+
+    /**
+     * Constructor
+     * @param fileDir The path to the directory
+     * @param file The file name
+     */
+    public Storage(String fileDir, String file) {
+        this.directory = fileDir;
+        this.file = file;
+        this.fileTasks = new ArrayList<>();
+    }
+
+    /**
+     * reads file. If file does not exist, create file
+     * @return List containing all tasks in file originally
+     */
+    public ArrayList<Task> loadData() throws DukeException {
+        // Make directory and/or file if they don't exist
+        File dataFolder = new File(directory);
+        boolean folderCreated = dataFolder.mkdirs();
+        File dataFile = new File(directory + "/" + file);
+        try {
+            boolean fileCreated = dataFile.createNewFile();
+        } catch (IOException e) {
+            System.out.println("Failed to create a new file");
+            return null;
+        }
+
+        // list to contain all tasks ass understood by bot.
+        ArrayList<Task> fileList = new ArrayList<>();
+        try {
+            Scanner fileReader = new Scanner(dataFile);
+            fileTasks.clear();
+            while (fileReader.hasNextLine()) {
+                String rawData = fileReader.nextLine();
+                fileTasks.add(rawData);
+                String[] data = rawData.split(" \\| ");
+                String taskType = data[0];
+                boolean isDone = data[1].equals("1");
+                Task task = null;
+                switch (taskType) {
+                    case "T":
```

Comment:
Perhaps need to remove the indentation for case clauses? 🤔
And I noticed the same issue in several other places below.